### PR TITLE
Update to nuke 8.0.0

### PR DESCRIPTION
**Pull request by matt-richardson**
Dependabot tried to update nuke to 8.0.0 in https://github.com/OctopusDeploy/OctoVersion/pull/190, which didn't work so well.

This does that upgrade, but also runs `nuke :update`

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -14,12 +14,18 @@ TEMP_DIRECTORY="$SCRIPT_DIR/.nuke/temp"
...
-DOTNET_CHANNEL="Current"
+DOTNET_CHANNEL="STS"
 
 export DOTNET_CLI_TELEMETRY_OPTOUT=1
+export DOTNET_NOLOGO=1
 export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
 export DOTNET_MULTILEVEL_LOOKUP=0
 
+
+###########################################################################
```
Comment (matt-richardson) [inline build.sh:25]: The custom code in the `configuration` section (as it was previously), makes `nuke :update` go :boom:.

Moving it under `execution` works (based on advice from @matkoch)

File: build/Build.cs [hunk 2]
```diff
@@ -73,7 +73,7 @@ class Build : NukeBuild
                 .SetProjectFile(Solution)
                 .SetConfiguration(Configuration)
                 .SetLoggers("trx")
-                .SetVerbosity(DotNetVerbosity.Normal)
+                .SetVerbosity(DotNetVerbosity.normal)
```
Comment (matt-richardson) [inline build/Build.cs:76]: Nuke changelog [says this is fixed](https://github.com/nuke-build/nuke/blob/59bdf80fcbe9b11f9e443a701d455b5aaa58c532/CHANGELOG.md?plain=1#L21) in 8.0.0, but... it appears to need to be lowercase still. :shrug:

File: build.ps1 [hunk 3]
```diff
@@ -18,10 +18,10 @@ $TempDirectory = "$PSScriptRoot\\.nuke\temp"
 
 $DotNetGlobalFile = "$PSScriptRoot\\global.json"
 $DotNetInstallUrl = "https://dot.net/v1/dotnet-install.ps1"
-$DotNetChannel = "Current"
+$DotNetChannel = "STS"
```
[the remark this answers is not in the document]
Comment (matt-richardson) [reply build.ps1:21]: This is a change made by the `nuke :update` - we dont really own this file to make any changes

File: build.ps1 [hunk 4]
```diff
@@ -61,9 +61,15 @@ else {
...
     $env:DOTNET_EXE = "$DotNetDirectory\dotnet.exe"
+    $env:PATH = "$DotNetDirectory;$env:PATH"
 }
 
 Write-Output "Microsoft (R) .NET SDK version $(& $env:DOTNET_EXE --version)"
 
+if (Test-Path env:NUKE_ENTERPRISE_TOKEN) {
+    & $env:DOTNET_EXE nuget remove source "nuke-enterprise" > $null
+    & $env:DOTNET_EXE nuget add source "https://f.feedz.io/nuke/enterprise/nuget" --name "nuke-enterprise" --username "PAT" --password $env:NUKE_ENTERPRISE_TOKEN > $null
+}
```
[the remark this answers is not in the document]
Comment (matt-richardson) [reply build.ps1:72]: This is a change made by the nuke :update

**Review by cailyoung — APPROVED** [review]

**Pull request merged**